bot/src/rpc_client_utils.rs

```diff
@@ -338,31 +340,61 @@ pub fn get_vote_account_info(
     ))
 }
 
-pub fn get_all_stake(
+pub fn get_all_stake_with_retry(
+    rpc_client: &RpcClient,
+    authorized_staker: Option<Pubkey>,
+) -> Result<Vec<(Pubkey, Account)>, Box<dyn error::Error>> {
+    let filters = authorized_staker.map(|s| {
+        vec![rpc_filter::RpcFilterType::Memcmp(rpc_filter::Memcmp {
+            offset: 12,
+            bytes: rpc_filter::MemcmpEncodedBytes::Base58(s.to_string()),
+            encoding: Some(rpc_filter::MemcmpEncoding::Binary),
+        })]
+    });
+    let config = RpcProgramAccountsConfig {
+        filters,
+        account_config: RpcAccountInfoConfig {
+            encoding: Some(solana_account_decoder::UiAccountEncoding::Base64),
+            commitment: Some(rpc_client.commitment()),
+            ..RpcAccountInfoConfig::default()
+        },
+        ..RpcProgramAccountsConfig::default()
+    };
+    let mut retries = 5u8;
+    let retry_delay = Duration::from_secs(10);
+    loop {
+        if retries == 0 {
+            return Err("Exhausted retries; connection to server is unhealthy".into());
+        }
+        let maybe_accounts =
+            rpc_client.get_program_accounts_with_config(&stake::program::id(), config.clone());
+
+        match maybe_accounts {
+            Err(ClientError {
+                kind: ClientErrorKind::Reqwest(err),
+                ..
+            }) if reqwest::Error::is_timeout(&err) => {
+                retries = retries.saturating_sub(1);
```

Comment:
this is gonna loop forever if we hit some other http error.  can switch to a for-loop+break and move the `if retries == 0` block to immediately after the loop

Reply:
Shouldn't it return the error? That's the intent with the catch-all underneath it, unless I did it wrong

Reply:
I tested this just to be sure, and this will properly fall through on any other error

Reply:
oh haha, dammit i had a pending comment from weeks ago here that i apparently never submitted the approval review for :facepalm: 

got confused by the hiding `Ok(_) | Err(_)` in `_ => { ... }`